language/move-prover/bytecode/src/livevar_analysis.rs

```diff
@@ -252,7 +276,9 @@ impl<'a> LiveVarAnalysis<'a> {
                     // Drop this load/assign as it is not used.
                 }
                 Bytecode::Call(attr_id, dests, oper, srcs, aa)
-                    if code_offset + 1 < code.len() && dests.len() == 1 =>
+                    if code_offset + 1 < code.len()
+                        && dests.len() == 1
+                        && !matches!(oper, Operation::BorrowLoc | Operation::BorrowGlobal(..)) =>
```

Comment:
Unfortunately, this is the culprit for the lengthy code in the `exp` files in `move-prover/bytecode/tests`.

And either
1) we keep this (i.e., drop the optimization on immediately dead borrows), such that we maintain the invariant that the `x` and `y` in `IsParent[x](y)` are both references, or
2) we support `IsParent[LocalRoot(x)](y)` and `IsParent[GlobalRoot(x)](y)`...

To me 1) is preferred due to simplicity in code and in concept, but let me know if we want to go to route 2).